src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py

```diff
@@ -52,34 +52,31 @@ def wrapper(*args, **kwargs):
 
 
 class AdversarialSimulator:
-    def __init__(self, *, azure_ai_project: Dict[str, Any]):
+    def __init__(self, *, azure_ai_project: Dict[str, Any], credential=None):
         """
         Initializes the adversarial simulator with a project scope.
 
         :param azure_ai_project: Dictionary defining the scope of the project. It must include the following keys:
             - "subscription_id": Azure subscription ID.
             - "resource_group_name": Name of the Azure resource group.
             - "project_name": Name of the Azure Machine Learning workspace.
-            - "credential": Azure credentials object for authentication.
+        :param credential: The credential for connecting to Azure AI project.
+        :type credential: TokenCredential
         :type azure_ai_project: Dict[str, Any]
         """
         # check if azure_ai_project has the keys: subscription_id, resource_group_name, project_name, credential
-        if not all(
-            key in azure_ai_project for key in ["subscription_id", "resource_group_name", "project_name", "credential"]
-        ):
-            raise ValueError(
-                "azure_ai_project must contain keys: subscription_id, resource_group_name, project_name, credential"
-            )
+        if not all(key in azure_ai_project for key in ["subscription_id", "resource_group_name", "project_name"]):
+            raise ValueError("azure_ai_project must contain keys: subscription_id, resource_group_name, project_name")
         # check the value of the keys in azure_ai_project is not none
-        if not all(
-            azure_ai_project[key] for key in ["subscription_id", "resource_group_name", "project_name", "credential"]
-        ):
-            raise ValueError("subscription_id, resource_group_name, project_name, and credential must not be None")
+        if not all(azure_ai_project[key] for key in ["subscription_id", "resource_group_name", "project_name"]):
+            raise ValueError("subscription_id, resource_group_name and project_name must not be None")
+        if "credential" in azure_ai_project and not credential:
+            credential = azure_ai_project["credential"]
         self.azure_ai_project = azure_ai_project
         self.token_manager = ManagedIdentityAPITokenManager(
             token_scope=TokenScope.DEFAULT_AZURE_MANAGEMENT,
             logger=logging.getLogger("AdversarialSimulator"),
-            credential=self.azure_ai_project["credential"],
+            credential=credential,

```

Comment:
If credential is None what is the behavior ? Seem like earlier it was required to be part of azure_ai_project now it has default value of None

Reply:
Fixed it!